Game/Source/GDKTestGyms/GDKTestGymsGameInstance.cpp

```diff
@@ -21,19 +22,50 @@ void UGDKTestGymsGameInstance::OnStart()
 	}
 }
 
-bool UGDKTestGymsGameInstance::Tick(float DeltaSeconds)
+float UGDKTestGymsGameInstance::AddAndCalcFps(int64 NowReal, float DeltaS)
 {
-	float Alpha = 0.8f;
-	float CurrFPS = 1.0f / DeltaSeconds; // TODO - we should remove FApp::GetIdleTime() from DeltaSeconds once the NFR regex is fixed
-	AverageFPS = (Alpha * AverageFPS) + ((1.0f - Alpha) * CurrFPS);
+	int64 DeltaTicks = FTimespan::FromSeconds(DeltaS).GetTicks();
+	TicksForFPS.Push(TPair<int64, int64>(NowReal, DeltaTicks)); 
+	TickWindowTotal += DeltaTicks;
+
+	int64 MinRange = NowReal - FTimespan::FromMinutes(2.0).GetTicks();
+	
+	// Trim samples
+	for(int i = 0; i < TicksForFPS.Num(); i++)
+	{
+		const FPSTimePoint& TimePoint = TicksForFPS[i];
+		if (TimePoint.Key > MinRange) // Find the first valid sample
+		{
+			int NumToRemove = i;
+			if (NumToRemove > 0)
+			{
+				for (int j = 0; j < NumToRemove; j++)
+				{
+					TickWindowTotal -= TicksForFPS[j].Value;
+				}
+				memmove(&TicksForFPS[0], &TicksForFPS[NumToRemove], (TicksForFPS.Num()-NumToRemove) * sizeof(FPSTimePoint));
```

Comment:
I wonder if we can remove the elements from TicksForFPS using TArray members.
This might allow us to remove the: TicksForFPS.SetNum(TicksForFPS.Num() - NumToRemove);

Reply:
It does but unfortunately can only remove one at a time, if we ever need to remove more than 1 we incur the cost of moving all the array elements multiple times (at least I couldn't see an API that removes a range). Perhaps a better data structure but that has its own complexity overheads (btree rebalancing) - since we only do this once per frame it may not be worth optimizing away too much, we could rearrange it to only remove when we calculate the FPS

Reply:
Yeah, if there isn't anything nice on TArray then I am completely happy!